### ISSUE TITLE
Style/RedundantCondition makes invalid autocorrect with condition on operator

### ISSUE THREAD

**Issue by layerssss** [rubocop/rubocop#10833]
`Style/RedundantCondition` makes an autocorrect with `-a` introducing side-effect for

```
        seconds = if video_started_at
          timestamp - video_started_at
        else
          timestamp - played_at
        end
```

autocorrected:

```
        seconds = (timestamp - video_started_at) || played_at
```

side-effect introduced: the original code wouldn't raise error when `video_started_at` is `nil`. The autocorrected one does raise `TypeError: can't convert nil into an exact number`

--------

## Expected behavior

Rubocop should indicate this can be autocorrected with `-A`

## Actual behavior

Rubocop indicates this can be autocorrected with `-a`

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.32.0 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 2.6.6 x86_64-linux)          │~
  - rubocop-rails 2.15.2                                                                       │~
  - rubocop-rspec 2.12.1
```
